File: src/frogger/screen/frame/elements/car/Car.java
package frogger.screen.frame.elements.car;

import frogger.screen.frame.elements.TexturedElement;

/**
 * @author Lucas Hagen.
 */
public class Car extends TexturedElement {

    private static String TEXTURE = "car.png";
    private static int WIDTH = 49;
    private static int HEIGHT = 24;

    public Car() {
        super(WIDTH, HEIGHT);

        setTexture(getClass().getResource(TEXTURE).toString(), WIDTH, HEIGHT);
    }

    public void move(int x, int y) {
        setTranslateX(getTranslateX() + x);
        setTranslateY(getTranslateY() + y);
    }

}

File: src/frogger/screen/frame/elements/car/Car.scala
package frogger.screen.frame.elements.car

import frogger.screen.frame.elements.TexturedElement
import frogger.screen.frame.helpers.PositionAndImageVariables

import scala.collection.mutable


abstract class Car extends TexturedElement(49, 24) {

  def setTextureOfCar() {}

  def setTranslateY(positionAndImages: PositionAndImageVariables): Double = {
    val start = (positionAndImages.H - 200).toInt
    val end: Int = positionAndImages.UP_MARGIN
    val r = new scala.util.Random
    val interval = start - end
    val r1 = start - r.nextInt(interval)
    val finalPosition = getPos(positionAndImages.carPositions.elements)
    setTranslateY(finalPosition)
    return finalPosition

  }

  def getPos(list: mutable.MutableList[Car]): Double = {
    if (list.isEmpty) {
      return 402
    }
    else {
      val lastElement = list(list.length - 1)
      return lastElement.localToScene(lastElement.getBoundsInLocal).getMinY - 64
    }
  }
}
